clients/unshaded/src/main/java/tachyon/client/netty/NettyRemoteBlockReader.java

```diff
@@ -0,0 +1,132 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.buffer.PooledByteBufAllocator;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+
+import tachyon.Constants;
+import tachyon.client.RemoteBlockReader;
+import tachyon.conf.TachyonConf;
+import tachyon.network.ChannelType;
+import tachyon.network.NettyUtils;
+import tachyon.network.protocol.RPCBlockRequest;
+import tachyon.network.protocol.RPCBlockResponse;
+import tachyon.network.protocol.RPCMessage;
+import tachyon.network.protocol.RPCMessageDecoder;
+import tachyon.network.protocol.RPCMessageEncoder;
+import tachyon.network.protocol.RPCResponse;
+
+/**
+ * Read data from remote data server using Netty.
+ */
+public final class NettyRemoteBlockReader implements RemoteBlockReader {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  // Share both the encoder and decoder with all the client pipelines.
+  private static final RPCMessageEncoder ENCODER = new RPCMessageEncoder();
+  private static final RPCMessageDecoder DECODER = new RPCMessageDecoder();
+
+  private static final TachyonConf TACHYON_CONF = new TachyonConf();
+  private static final ChannelType CHANNEL_TYPE = TACHYON_CONF.getEnum(
+      Constants.USER_NETTY_CHANNEL, ChannelType.defaultType());
+  private static final Class<? extends SocketChannel> CLIENT_CHANNEL_CLASS = NettyUtils
+      .getClientChannelClass(CHANNEL_TYPE);
+  // Reuse EventLoopGroup for all clients.
+  // Use daemon threads so the JVM is allowed to shutdown even when daemon threads are alive.
+  private static final EventLoopGroup WORKER_GROUP = NettyUtils.createEventLoop(CHANNEL_TYPE,
+      TACHYON_CONF.getInt(Constants.USER_NETTY_WORKER_THREADS, 1), "netty-client-worker-%d", true);
+
+  private final Bootstrap mClientBootstrap;
+  private final ClientHandler mHandler;
+
+  // TODO: Creating a new remote block reader may be expensive, so consider a connection pool.
+  public NettyRemoteBlockReader() {
+    mHandler = new ClientHandler();
+    mClientBootstrap = createClientBootstrap();
+  }
+
+  @Override
+  public ByteBuffer readRemoteBlock(String host, int port, long blockId, long offset, long length)
+      throws IOException {
+    InetSocketAddress address = new InetSocketAddress(host, port);
+
+    try {
+      ChannelFuture f = mClientBootstrap.connect(address).sync();
+
+      LOG.info("Connected to remote machine " + address);
+      Channel channel = f.channel();
+      SingleResponseListener listener = new SingleResponseListener();
+      mHandler.addListener(listener);
+      channel.writeAndFlush(new RPCBlockRequest(blockId, offset, length));
+
+      RPCResponse response = listener.get(1, TimeUnit.SECONDS);
```

Comment:
the timeout value (1 sec) is hardcoded here, better to have it configured somewhere, e.g., a final static int defined in this class.